src/crawlers/basic_crawler.js

```diff
@@ -31,6 +32,7 @@ const DEFAULT_OPTIONS = {
         log.error('BasicCrawler: Request failed and reached maximum retries', details);
     },
     autoscaledPoolOptions: {},
+    sessionPoolOptions: {}, // We could add sessionPool true/false config to use/not use SessionPool.
```

Comment:
I think it makes sense, I wouldn't force usage of session pool all the time. How about calling it `useSessionPool` ?